firmware/application/ui_navigation.cpp

```diff
@@ -297,7 +304,9 @@ void SystemStatusView::set_title(const std::string new_value) {
     if (new_value.empty()) {
         title.set(default_title);
     } else {
-        title.set(new_value);
+        // Limit length of title string to prevent partial characters if too many StatusView icons
+        size_t max_len = (240 - 16 - 7 - status_icons.width()) / 8;
```

Comment:
Quite a few magic constants. Could you use the title’s parent_rect left() instead? There’s also a screen_width constant you can use. 

Really the max could just be like the status tray’s left - the title’s left.